core/src/main/scala/org/ensime/indexer/SearchService.scala

```diff
@@ -301,7 +368,7 @@ class IndexingQueueActor(searchService: SearchService) extends Actor with ActorL
   // De-dupes files that have been updated since we were last told to
   // index them. No need to aggregate values: the latest wins. Key is
   // the URI because FileObject doesn't implement equals
-  var todo = Map.empty[String, FileObject]
+  val todo = new m.HashMap[FileObject, m.Set[FileObject]] with m.MultiMap[FileObject, FileObject]
```

Comment:
oh this is going to cause problems... I just noticed now that you use `FileObject` as the key. Read the comment ;-) Definitely a bug in here.